src/nc/core/likec/Simplifier.cpp

```diff
@@ -400,7 +419,48 @@ std::unique_ptr<MemberAccessOperator> Simplifier::simplify(std::unique_ptr<Membe
 
 std::unique_ptr<Expression> Simplifier::simplify(std::unique_ptr<Typecast> node) {
     node->operand() = simplify(std::move(node->operand()));
-
+    /*
+     * simplify casts within casts
+     * woooooo! this works!
+     * and, it has a fairly noticeable impact
+     */
+    if(node->operand()->is<UnaryOperator>()
+                                          && node->type()->isInteger() &&
+            node->operand()->as<UnaryOperator>()->operatorKind() == UnaryOperator::DEREFERENCE) {
```

Comment:
Could you fix the indentation (not only here, but everywhere: indent=4, use spaces, not tabs)?